backend/controllers/userController.js

```diff
@@ -0,0 +1,61 @@
+const asyncHandler = require("express-async-handler");
+const generateToken = require("../utils/jsonUtils");
+const User = require("../models/userModel");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken")
+
+const registerUser = asyncHandler(async (req, res) => {
+  console.log("HERE");
+  const { name, email, password } = req.body;
+
+  const userExists = await User.findOne({ email });
+
+  if (userExists) {
+    res.status(400); // bad request
+    throw new Error("User already exists");
+  }
+
+  const user = await User.create({
+    name: name,
+    email: email,
+    password: password
+  });
+
+  if (user) {
+    res.status(202).json({
```

Comment:
I would suggest statusCode 201 as the user would already be successfully created

Reply:
Fixed: Changes are reflected in recent commits